app/models/bounce.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+#  Copyright (c) 2025-2025, Schweizer Wanderwege. This file is part of
+#  hitobito and licensed under the Affero General Public License version 3
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito.
+
+class Bounce < ApplicationRecord
+  class UnexpectedBlock < StandardError
+    def initialize(mail, msg = nil)
+      super(msg || "The checked mail (#{mail}) was blocked, but not expected to be.")
+    end
+  end
+
+  BLOCK_THRESHOLD = 5
+
+  scope :blocked, -> { where.not(blocked_at: nil) }
+  scope :of_mailing_list, ->(id) {
+    # && is "set union" in postgresql, so it is true if there is an overlap
+    # between the param and the stored array.
+    where("mailing_list_ids && '{:mailing_list_id}'", mailing_list_id: id)
+  }
+
+  before_validation :evaluate_blocking
+
+  class << self
+    def record(email, mailing_list_id: nil)
+      bounce = find_or_create_by(email: email)
+      if mailing_list_id.present?
```

Comment:
This duplicates the mailing_list_ids in the array... Idk if that is the desired behavior, but it seems odd:
`mailing_list_ids: [1, 1, 1, 1, 1]`